server/routes/courtRegister/amendCourtBuildingValidator.ts

```diff
@@ -17,50 +10,31 @@ export default async function validate(
   courtLookup: (subCode: string) => Promise<Court | null>,
   courtBuildingLookup: (subCode: string) => Promise<CourtBuilding | null>
 ): Promise<string> {
-  const errors: Array<{ text: string; href: string }> = []
-
-  if (isBlank(form.buildingname)) {
-    errors.push({ text: 'Enter the building name', href: '#buildingname' })
-  }
-
-  if (isBlank(form.addressline1)) {
-    errors.push({ text: 'Enter the first line of the address', href: '#addressline1' })
-  }
-
-  if (isBlank(form.addresstown)) {
-    errors.push({ text: 'Enter the town or city', href: '#addresstown' })
-  }
-
-  if (isBlank(form.addresscounty)) {
-    errors.push({ text: 'Enter the county', href: '#addresscounty' })
-  }
-
-  if (isBlank(form.addresspostcode)) {
-    errors.push({ text: 'Enter the postcode, like AA11AA', href: '#addresspostcode' })
-  } else if (!isValidPostcode(form.addresspostcode)) {
-    errors.push({ text: 'Enter a real postcode, like AA11AA', href: '#addresspostcode' })
-  }
-
-  if (isBlank(form.addresscountry)) {
-    errors.push({ text: 'Select the country', href: '#addresscountry' })
-  }
-
-  if (!isBlank(form.subCode)) {
-    const existingCourt = await courtLookup(form.subCode)
-    if (existingCourt) {
-      errors.push({ text: `${existingCourt.courtName} already has that code. Choose another code`, href: '#subCode' })
-    } else {
-      const existingCourtBuilding = await courtBuildingLookup(form.subCode)
-      if (existingCourtBuilding) {
-        if (existingCourtBuilding.id !== Number.parseInt(form.id, 10)) {
-          errors.push({
-            text: `The court building ${existingCourtBuilding.buildingName} already has that code. Choose another code`,
-            href: '#subCode',
-          })
-        }
-      }
+  const errors = await validateAsync(
+    form,
+    {
+      buildingname: 'required',
+      addressline1: 'required',
+      addresstown: 'required',
+      addresscounty: 'required',
+      addresspostcode: ['required', 'postcode'],
+      addresscountry: 'required',
+      subCode: ['between:0,6', `unique-subcode:${form.id}`],
```

Comment:
does the framework eval this? or is it a mini embedded dsl?

Reply:
For their inbuilt ones that parse the stuff after the colon. For custom you will receive and extra parameter with the righthand side of the colon. It is up to you to decide what it means.